packages/react-components/react-avatar/src/components/AvatarGroupItem/useAvatarGroupItemStyles.ts

```diff
@@ -1,42 +1,97 @@
+import { useSizeStyles } from '../../Avatar';
 import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
-import { tokens } from '@fluentui/react-theme';
+import { tokens, typographyStyles } from '@fluentui/react-theme';
+import type { AvatarSizes } from '../../Avatar';
 import type { AvatarGroupProps } from '../../AvatarGroup';
 import type { AvatarGroupItemSlots, AvatarGroupItemState } from './AvatarGroupItem.types';
-import type { AvatarSizes } from '../../Avatar';
 import type { SlotClassNames } from '@fluentui/react-utilities';
-import { useContextSelector } from '@fluentui/react-context-selector';
-import { AvatarGroupContext } from '../../contexts/AvatarGroupContext';
 
 export const avatarGroupItemClassNames: SlotClassNames<AvatarGroupItemSlots> = {
   root: 'fui-AvatarGroupItem',
   avatar: 'fui-AvatarGroupItem__avatar',
   overflowLabel: 'fui-AvatarGroupItem__overflowLabel',
 };
 
+const avatarGroupItemDividerWidthVar = '--fuiAvatarGroupItem__divier--width';
+
 /**
  * Styles for the root slot
  */
 const useRootStyles = makeStyles({
   base: {
     alignItems: 'center',
-    display: 'flex',
+    display: 'inline-flex',
+    flexShrink: 0,
     position: 'relative',
   },
   overflowItem: {
-    ...shorthands.padding(tokens.spacingVerticalXS, tokens.spacingVerticalXS),
+    ...shorthands.padding(tokens.spacingVerticalXS, tokens.spacingHorizontalXS),
   },
   nonOverflowItem: {
     ...shorthands.borderRadius(tokens.borderRadiusCircular),
   },
 });
 
+/**
+ * Styles for the avatar slot
+ */
+const useAvatarStyles = makeStyles({
+  nonOverflowItem: {
+    position: 'absolute',
+  },
+  pie: {
+    ...shorthands.borderRadius(0),
+  },
+});
+
 /**
  * Styles for the label slot
  */
 const useOverflowLabelStyles = makeStyles({
-  overflowItem: {
+  base: {
     marginLeft: tokens.spacingHorizontalS,
+    color: tokens.colorNeutralForeground1,
+    ...typographyStyles.body1,
+  },
+});
+
+/**
+ * Styles for the pie layout
+ */
+const usePieStyles = makeStyles({
+  base: {
+    position: 'absolute',
+  },
+  twoPies: {
+    '&:first-child': {
+      clipPath: `inset(0 calc(25% + (var(${avatarGroupItemDividerWidthVar}) / 2)) 0 25%)`,
+      left: '-25%',
+    },
+    '&:nth-child(2)': {
+      clipPath: `inset(0 25% 0 calc(25% + (var(${avatarGroupItemDividerWidthVar}) / 2)))`,
+      left: '25%',
+    },
+  },
+  threePies: {
+    '&:first-child': {
+      clipPath: `inset(0 calc(25% + (var(${avatarGroupItemDividerWidthVar}) / 2)) 0 25%)`,
+      left: '-25%',
+    },
+    '&:not(:first-child)': {
+      // Since the two AvatarGroupItems on the right are scaled by 0.5, the divider width should not be halved.
+      clipPath: `inset(0 0 var(${avatarGroupItemDividerWidthVar}) var(${avatarGroupItemDividerWidthVar}))`,
+      left: '50%',
+      transform: 'scale(0.5)',
+      transformOrigin: '0 0',
+    },
+    '&:nth-of-type(3)': {
```

Comment:
Is there a reason you're using `:nth-child` above, and `:nth-of-type` here? Ideally it would be consistent (probably `:nth-child` to go along with `:first-child`)